File: Napakalaki Ruby/pkg/NapakalakiRuby-0.0.1/lib/dice.rb
# By: Miguel Robles Urquiza
#     Jesús Sánchez de Lechina Tejada
 require "singleton"

module NapakalakiGame

  class Dice
     
    include Singleton

    def nextNumber
        return 1 + rand(6)
    end
    
  end
end

File: Napakalaki Java/src/NapakalakiGame/Dice.java
package NapakalakiGame;

/**
 *
 * @author Jesús Sánchez de Lechina Tejada & Miguel Ángel Robles Urquiza
 */

public class Dice {
    private static final Dice instance = new Dice();
    
    private Dice(){}
    
    public static Dice getInstance(){
        return instance;
    }
    
    public int nextNumber(){
        return (int) ((Math.random()*6)+1);
        
    }
}